test/e2e/persistent_volumes.go

```diff
@@ -465,78 +532,161 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
 
-	// Create a claim first, then a pre-bound nfs PV that matches the claim,
-	// and a pod that contains the claim. Verify that the PV and PVC bind
-	// correctly, and that the pod can write to the nfs volume.
-	It("create a PVC and a pre-bound PV: test write access [Flaky]", func() {
+			pv, pvc, err = createPVCPV(c, serverIP, ns, true)
+			if err != nil {
+				framework.Failf("%v", err)
+			}
 
-		pv, pvc, err = createPVCPV(c, serverIP, ns, true)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
+			// validate PV-PVC, create and verify writer pod, delete PVC
+			// and PV
+			pv, pvc, err = completeTest(f, c, ns, pv, pvc)
+			if err != nil {
+				framework.Failf("%v", err)
+			}
+		})
 
-		// validate PV-PVC, create and verify writer pod, delete PVC
-		// and PV
-		pv, pvc, err = completeTest(f, c, ns, pv, pvc)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
-	})
+		// Create a nfs PV first, then a pre-bound PVC that matches the PV,
+		// and a pod that contains the claim. Verify that the PV and PVC bind
+		// correctly, and that the pod can write to the nfs volume.
+		It("create a PV and a pre-bound PVC: test write access [Flaky]", func() {
 
-	// Create a nfs PV first, then a pre-bound PVC that matches the PV,
-	// and a pod that contains the claim. Verify that the PV and PVC bind
-	// correctly, and that the pod can write to the nfs volume.
-	It("create a PV and a pre-bound PVC: test write access [Flaky]", func() {
+			pv, pvc, err = createPVPVC(c, serverIP, ns, true)
+			if err != nil {
+				framework.Failf("%v", err)
+			}
 
-		pv, pvc, err = createPVPVC(c, serverIP, ns, true)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
+			// validate PV-PVC, create and verify writer pod, delete PVC
+			// and PV
+			pv, pvc, err = completeTest(f, c, ns, pv, pvc)
+			if err != nil {
+				framework.Failf("%v", err)
+			}
+		})
+	})
 
-		// validate PV-PVC, create and verify writer pod, delete PVC
-		// and PV
-		pv, pvc, err = completeTest(f, c, ns, pv, pvc)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
+	Context("with multiple PVs and PVCs", func() {
+
+		var pvs []*api.PersistentVolume
+		var pvcs []*api.PersistentVolumeClaim
+
+		AfterEach(func() {
+			if c != nil && len(ns) > 0 {
+				if pvcs != nil && len(pvcs) > 0 {
+					framework.Logf("PVC List is non-nil, deleting all PVCs and destroying list")
+					for _, pvc := range pvcs {
+						if pvc != nil && len(pvc.Name) > 0 {
+							if err := c.PersistentVolumeClaims(ns).Delete(pvc.Name); err != nil {
+								framework.Logf("Delete PVC %v failed: ", err)
+							}
+						}
+						if _, err := c.PersistentVolumeClaims(ns).Get(pvc.Name); err == nil {
+							framework.Failf("Expected PVC to be nil after delete but got non-nil value")
+						}
+					}
+					pvcs = nil
+				}
+				if pvs != nil && len(pvs) > 0 {
+					framework.Logf("PV List is non-nil, deleting all PVs and destroying list")
+					for _, pv := range pvs {
+						if pv != nil && len(pv.Name) > 0 {
```

Comment:
Here you check for pv != nil (again implying a sparse slice) but you didn't check for nil on L124.